### DISCUSSION TITLE
Custom Connectors: Support custom connect data.

### DISCUSSION THREAD

**Discussion by johanneskares** [wevm/wagmi#4441] [Ideas]
There are several usecases for custom connectors that need custom data before connection.

#### Example A: A headless E-Mail wallet (like https://magic.link)
```typescript
await connectAsync({ connector: magicWallet, customData: { email, onReadyForOtpEnter: async () => { ... } })
```

#### Example B: Coinbase Wallet custom singup
As supported in this PR: https://github.com/wevm/wagmi/pull/4440
```typescript
await connectAsync({ connector: coinbaseWallet, customData: { onboarding: "instant" })
```

#### Custom Connectors should be able to implement
```typescript
async connect({ chainId, customData } = {}) {
   // use customData
    ....
}
```

**Comment by tmm** [comment]
https://github.com/wevm/wagmi/pull/4453

**Reply by johanneskares** [reply]
amazing, thanks for implementing this!
